command_line_assistant/utils/cli.py

```diff
@@ -214,3 +236,140 @@ def create_subparser(parser: SubParsersAction, name: str, help: str) -> Argument
     )
 
     return custom_parser
+
+
+def command(
```

Comment:
A usage example would help illustrate the objective of this decorator, plus the following `@argument` decorator and `Command` class. Maybe at the top of the file, since all the types in the file are involved in providing the `@command` and `@argument` decorator design pattern.